fe/fe-core/src/main/java/org/apache/doris/nereids/analyzer/UnboundAlias.java

```diff
@@ -89,4 +89,8 @@ public UnboundAlias withChildren(List<Expression> children) {
     public Optional<String> getAlias() {
         return alias;
     }
+
+    public void setAlias(Optional<String> alias) {
+        this.alias = alias;
+    }
```

Comment:
this is a mistake of `UnboundAlias`. In Nereids all Expression and plan should be Immutable. So we should let alias be final. and add a function `withAlias` like `withChildren` to generate a new `UnboundAlias` object with new alias